### PR TITLE
Merge in changes to tests from ruby/test/openssl.

### PR DESCRIPTION
**Pull request by ioquatix**
@mame @hsbt @nobu There is one issue in `test_sysread_nonblock_and_syswrite_nonblock_keywords` but I'm not sure how to fix it yet. Do you have any ideas?

### CODE REVIEW

File: test/test_ssl.rb [hunk 1]
```diff
@@ -206,8 +206,20 @@ def test_sysread_and_syswrite
     }
   end
 
+  def test_sysread_nonblock_and_syswrite_nonblock_keywords
```
Comment (ioquatix) [inline test/test_ssl.rb]: @jeremyevans do you mind letting me know what the point of this test was and how we can make it work more reliably?

Comment (ioquatix) [reply test/test_ssl.rb]: Also, do we expose `sysread_nonblock` and `syswrite_nonblock` because those seem counter intuitive.

**Comment by jeremyevans** [comment]
It was added in https://github.com/ruby/ruby/commit/3959469f240edb6c1f43976bbb72a0ba9105a6b1.  It just exists to make sure there are no keyword argument separation issues in the code path where `syswrite_nonblock` or `sysread_nonblock` are called before the SSL connection is made.

**Comment by ioquatix** [comment]
Calling those functions before the SSL connection is made seems invalid. @jeremyevans is this a spec for keyword arguments or OpenSSL behaviour? (or both?)

**Comment by jeremyevans** [comment]
It is a spec for keyword arguments.  There was already a warning in this code path, so this is not a code path that should be hit normally.

**Comment by ioquatix** [comment]
Okay, on that basis, I'm not going to merge it, unless you believe there is some other reason why we should keep it. Thanks!

**Comment by ioquatix** [comment]
In conclusion, the test is out of scope for OpenSSL, especially because it depends on a code path that generates a warning in normal situations.

**Pull request merged**